pkg/controller/computenodeopenstack/computenodeopenstack_controller.go

```diff
@@ -1229,29 +1268,43 @@ func deleteAllBlockerPodFinalizer(r *ReconcileComputeNodeOpenStack, instance *co
 		log.Info(fmt.Sprintf("Post draining job completed for node: %s", node.Name))
 	}
 	if !allNodesPostDrained {
-		return fmt.Errorf("Not all Post NodeDrain jobs finished, reconciling")
+		// if not all Post draining jobs finished, return a StatusReasonAlreadyExists.
+		// AlreadyExists error will be ignored as we want to wait for the job event to
+		// not spam the operator log with too many not needed reconcile requrest.
+		return &errors.StatusError{ErrStatus: metav1.Status{
+			Status: metav1.StatusFailure,
+			Code:   http.StatusConflict,
+			Reason: metav1.StatusReasonAlreadyExists,
+			Details: &metav1.StatusDetails{
+				Group: instance.GroupVersionKind().Group,
+				Kind:  instance.Kind,
+				Name:  instance.Name,
+			},
+			Message: fmt.Sprintf("Not all POST NodeDrain jobs finished"),
+		}}
 	}
 	log.Info(fmt.Sprintf("All Post NodeDrain jobs finished"))
 
-	// 3. Cleanup finished POST drain jobs
+	// 3. Cleanup
 	for _, node := range nodeList.Items {
-		log.Info(fmt.Sprintf("Deleting draining jobs for node: %s", node.Name))
+		// delete blocker pods to proceed with node removal from OCP
+		log.Info(fmt.Sprintf("Deleting blocker pod finalizer for node: %s", node.Name))
+		err = deleteBlockerPodFinalizer(r.client, node.Name, instance)
```

Comment:
I think this needs to be executed after the deleteJobWithName... otherwise there may be leftovers

Reply:
as discussed there were issues in jobs being recreated, added a check if the finalizer is already deleted and scipt the node, like in nodestatus update.